lib/model/QubitPhysicalObject.php

```diff
@@ -326,4 +326,86 @@ public function getRelationsAsArray(int $typeId = null)
 
     return $results->fetchAll(PDO::FETCH_ASSOC);
   }
+
+  /**
+   * An quick update of an existing physical_object database row
+   *
+   * If passed data wouldn't alter the current db data, return false without
+   * writing to the database
+   *
+   * Currently only updates the following columns:
+   * - object.updated_at
+   * - physical_object.type_id
+   * - physical_object_i18n.location
+   *
+   * @param array $data a keyed array (column => value)
+   * @param PropelPDO $connection a connection object
+   */
+  public function quickUpdate(array $data, $connection = null)
+  {
+    $doUpdate = false;
+
+    $updateCols = [
+      'typeId',
+      'location',
+    ];
+
+    if (!isset($connection))
+    {
+      $connection = QubitTransactionFilter::getConnection();
+    }
+
+    foreach ($updateCols as $name)
+    {
+      if (!empty($data[$name]) && $data[$name] != $this->$name)
+      {
+        $doUpdate = true;
+        $newvals[$name] = $data[$name];
```

Comment:
Can't see where this array gets initialized...

Reply:
Good catch, I'll add a line to initialize `$newvals` it at the top of the method.